lib/manifold/api/project.rb

```diff
@@ -27,7 +36,11 @@ def vectors_directory
 
       private
 
-      attr_writer :name, :directory
+      def workspace_directories
+        workspaces_directory.children.select(&:directory?)
+      end
+
+      attr_writer :name, :logger, :directory
```

Comment:
Grr. `attr_writer` and `attr_accessor` should be removed from the language.

The _only_ place you are setting these is in the constructor, so you've created a whole bunch of typing for yourself for no conceivable reason.

Remove this and change your constructor:
```ruby
def initialise(name, logger: Logger.new($stdout), directory: Pathname.pwd.join(name))
  @name = name
  @logger = logger
  @directory = Pathname(directory)
end
```
Just like the good Matz intended.

Reply:
Sounds good — still grokking the mechanics around those `attr_*` utilities.